python/samples/getting_started/workflows/orchestration/handoff_specialist_to_specialist.py

```diff
@@ -0,0 +1,224 @@
+
+    for event in events:
+        if isinstance(event, WorkflowStatusEvent) and event.state in {
+            WorkflowRunState.IDLE,
+            WorkflowRunState.IDLE_WITH_PENDING_REQUESTS,
+        }:
+            print(f"[status] {event.state.name}")
+
+        elif isinstance(event, WorkflowOutputEvent):
+            conversation = cast(list[ChatMessage], event.data)
+            if isinstance(conversation, list):
+                print("\n=== Final Conversation ===")
+                for message in conversation:
+                    # Filter out messages with no text (tool calls)
+                    if not message.text.strip():
+                        continue
+                    speaker = message.author_name or message.role.value
+                    print(f"- {speaker}: {message.text}")
+                print("==========================")
+
+        elif isinstance(event, RequestInfoEvent):
+            if isinstance(event.data, HandoffUserInputRequest):
+                _print_handoff_request(event.data)
+            requests.append(event)
+
+    return requests
+
+
+def _print_handoff_request(request: HandoffUserInputRequest) -> None:
+    """Display a user input request with conversation context."""
+    print("\n=== User Input Requested ===")
+    # Filter out messages with no text for cleaner display
+    messages_with_text = [msg for msg in request.conversation if msg.text.strip()]
+    print(f"Last {len(messages_with_text)} messages in conversation:")
+    for message in messages_with_text[-5:]:  # Show last 5 for brevity
+        speaker = message.author_name or message.role.value
+        text = message.text[:100] + "..." if len(message.text) > 100 else message.text
+        print(f"  {speaker}: {text}")
+    print("============================")
+
+
+async def main() -> None:
+    """Demonstrate specialist-to-specialist handoffs in a multi-tier support scenario.
+
+    This sample shows:
+    1. Triage agent routes to replacement specialist
+    2. Replacement specialist hands off to delivery specialist
+    3. Delivery specialist can hand off to billing if needed
+    4. All transitions are seamless without returning to user until complete
+
+    The workflow configuration explicitly defines which agents can hand off to which others:
+    - triage_agent → replacement_agent, delivery_agent, billing_agent
+    - replacement_agent → delivery_agent, billing_agent
+    - delivery_agent → billing_agent
+    """
+    chat_client = AzureOpenAIChatClient(credential=AzureCliCredential())
+    triage, replacement, delivery, billing = create_agents(chat_client)
+
+    # Configure multi-tier handoffs explicitly
+    # This allows specialists to hand off to other specialists
+    workflow = (
+        HandoffBuilder(
+            name="multi_tier_support",
+            participants=[triage, replacement, delivery, billing],
+        )
+        .starting_agent("triage_agent")
+        .with_handoffs({
```

Comment:
Since we are already using fluent API, maybe we can make use of it to make the method interface "shallower" and types clearer: 

```python
.add_handoff(triage_agent, [replacement_agent, delivery_agent]) # add multiple targets at a time.
.add_handoff(delivery_agent,  billing_agent) # add 1 target, but with optional parameters like customizing the handoff method name, handoff tool call response string, etc.
```